extensions/notebook/src/test/book/book.test.ts

```diff
@@ -44,20 +44,24 @@ describe('BookTreeViewProvider.getChildren', function (): void {
 	let book: BookTreeItem;
 
 	this.beforeAll(async () => {
-		fs.mkdirSync(rootFolderPath);
-		fs.mkdirSync(dataFolderPath);
-		fs.mkdirSync(contentFolderPath);
-		fs.writeFileSync(configFile, 'title: Test Book');
-		fs.writeFileSync(tableOfContentsFile, '- title: Notebook\n  url: /notebook\n- title: Markdown\n  url: /markdown\n- title: GitHub\n  url: https://github.com/\n  external: true');
-		fs.writeFileSync(notebookFile, '');
-		fs.writeFileSync(markdownFile, '');
-		mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
-		let folder: vscode.WorkspaceFolder = {
-			uri: vscode.Uri.parse(rootFolderPath),
-			name: '',
-			index: 0
-		};
-		bookTreeViewProvider = new BookTreeViewProvider([folder], mockExtensionContext.object);
+		try {
+			await fs.mkdir(rootFolderPath);
+			await fs.mkdir(dataFolderPath);
+			await fs.mkdir(contentFolderPath);
+			await fs.writeFile(configFile, 'title: Test Book');
+			await fs.writeFile(tableOfContentsFile, '- title: Notebook\n  url: /notebook\n- title: Markdown\n  url: /markdown\n- title: GitHub\n  url: https://github.com/\n  external: true');
+			await fs.writeFile(notebookFile, '');
+			await fs.writeFile(markdownFile, '');
+			mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
+			let folder: vscode.WorkspaceFolder = {
+				uri: vscode.Uri.parse(rootFolderPath),
+				name: '',
+				index: 0
+			};
+			bookTreeViewProvider = new BookTreeViewProvider([folder], mockExtensionContext.object);
+		} catch (e) {
+			console.log(e);
+		}
```

Comment:
I think an error here should still fail the test. Why are you ignoring the errors?

Reply:
Yeah please fail but with an assertion so it should be logged properly in the output and help us debug 😄

Reply:
Added an assertion!

Reply:
I'd just remove the try/catch - exceptions are going to fail the test anyways so no reason to wrap it an extra time

---
In reply to: [311765512](https://github.com/microsoft/azuredatastudio/pull/6633#discussion_r311765512) [](ancestors = 311765512)